linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/CoreJSLib.scala

```diff
@@ -1399,49 +1428,11 @@ private[emitter] object CoreJSLib {
 
       def isAssignableFrom = {
         val that = varRef("that")
-        val thatFakeInstance = varRef("thatFakeInstance")
         MethodDef(static = false, StringLiteral("isAssignableFrom"),
             paramList(that), {
-          If(genIdentBracketSelect(This(), "isPrimitive") ||
-              genIdentBracketSelect(that, "isPrimitive"), {
-            Return(This() === that)
-          }, {
-            Block(
-                genEmptyMutableLet(thatFakeInstance.ident),
-                If(that === genClassDataOf(BoxedStringClass), {
-                  thatFakeInstance := str("")
-                }, {
-                  If(that === genClassDataOf(BoxedBooleanClass), {
-                    thatFakeInstance := bool(false)
-                  }, {
-                    If({
-                      List(BoxedByteClass, BoxedShortClass, BoxedIntegerClass,
-                          BoxedFloatClass, BoxedDoubleClass).map { cls =>
-                        that === genClassDataOf(cls)
-                      }.reduceLeft(_ || _)
-                    }, {
-                      thatFakeInstance := int(0)
-                    }, {
-                      If(that === genClassDataOf(BoxedLongClass), {
-                        thatFakeInstance := genLongZero()
-                      }, {
-                        If(that === genClassDataOf(BoxedCharacterClass), {
-                          thatFakeInstance := genBoxedCharZero()
-                        }, {
-                          If(that === genClassDataOf(BoxedUnitClass), {
-                            thatFakeInstance := Undefined()
-                          }, {
-                            thatFakeInstance := ObjectConstr(List(classData -> that))
-                          })
-                        })
-                      })
-                    })
-                  })
-                }),
-                Return(Apply(genIdentBracketSelect(This(), "isInstance"),
-                    thatFakeInstance :: Nil))
-            )
-          })
+          Return(
+              (This() === that) || // fast path
+              Apply(This() DOT "isAssignableFromFun", that :: Nil))
```

Comment:
It seems that this distinction of `isAssignableFromFun` / `isAssignableFrom` has emerged because we currently DCE `isAssignableFrom` is `jl.Class` is not instantiated (because we do not use it internally). I'm not sure this is still warranted. Consider simply using the public field. (There seems to be precedent for that for `isInstance`).

If you do this, also initialize it in the `ctor`.

Reply:
> we never DCE isAssignableFrom.

WDYM? If `jl.Class` is not instantiated, we do not emit it. Look for `globalKnowledge.isClassClassInstantiated` in `CoreJSLib.scala`.

That being said, if we keep the separation between the private / public function, we can also keep it this way.

Reply:
Huh. I had completely forgotten about that. Yes, you're right, of course.

I had not taken that into account, but it's true that the private/public decomposition also addresses that issue. That said, in a better future we would also DCE `isInstance` if `!isClassClassInstantiated`, and then that problem would not exist (but we'd still want the decomposition for the fast path).